Query filters, array membership, bundles snippets for GA

markarndt opened the pull request.

markarndt committed:
Start roughing in array membership snippets for GA.

markarndt committed:
Fix vector declaration.

morganchen12 commented:
You'll need to update the binary source to point to the latest version of the SDK in order to fix the build: 

- https://github.com/firebase/snippets-cpp/blob/master/.github/workflows/firestore-android.yml#L17
- https://github.com/firebase/snippets-cpp/blob/master/.github/workflows/firestore-ios.yml#L23

markarndt committed:
Update SDK binary versions.

markarndt committed:
Update firestore-ios.yml

markarndt committed:
Update firestore-android.yml

Update SDK binary version.

markarndt committed:
Test change for new DocumentRefernce.AddSnapshotListener method signature.

markarndt committed:
Merging.

Merge branch 'mea-where-updates' of https://github.com/firebase/snippets-cpp into mea-where-updates

markarndt committed:
Correct all AddSnapshotListener calls. Try std::string and FieldValue::String.

markarndt committed:
Keep tinkering with vector init.

markarndt committed:
Try typename.

markarndt committed:
Oof. C++ types.

markarndt committed:
Oof. More C++ typing.

markarndt committed:
OK, that's syntactically correct, so replace in other query filter snippets.

morganchen12 committed:
fix iOS build

morganchen12 committed:
Fix android build and add bundle snippet

dconeybe commented on firestore/android/FirestoreSnippetsCpp/app/src/main/cpp/snippets.cpp:
```diff
@@ -855,6 +855,57 @@ void ReadDataQueryOperators(firebase::firestore::Firestore* db) {
+  // [START cpp_in_filter]
+  CollectionReference cities_ref = db->Collection("cities");
+
+  cities_ref.WhereIn("country", std::vector<typename firebase::firestore::FieldValue> {FieldValue::String("USA"), FieldValue::String("Japan")});
```
Can you omit the "firebase::firestore::" prefix here, since there is the "using" clause above? Applies elsewhere in this file as well.

morganchen12 replied:
Done.

dconeybe commented on firestore/android/FirestoreSnippetsCpp/app/src/main/cpp/snippets.cpp:
```diff
@@ -1061,6 +1112,54 @@ void ReadDataPaginateQuery(firebase::firestore::Firestore* db) {
+      }
+      case LoadBundleTaskProgress::State::kInProgress: {
+        std::cout << "Bytes loaded from bundle: " << progress.bytes_loaded()
+                  << '\n';
```
nit: use `std::endl` instead of `'\n'` for portability. Applies elsewhere in this file too.

morganchen12 replied:
Done.

morganchen12 committed:
clean up snippets

morganchen12 commented:
@googlebot I consent.
